lib/src/http/websocket_conn.dart

```diff
@@ -334,13 +336,22 @@ class WebSocketConnection extends Connection {
       } else if (encoded is List<int>) {
         dataOut += encoded.length;
       } else {
-        logger.warning(formatLogMessage("invalid data frame"));
+        logger.warning(formatLogMessage("Invalid data frame."));
       }
     }
+
+    if (socket.readyState != WebSocket.OPEN) {
+      logger.severe(
+        formatLogMessage("Error writing to socket: socket is closed.")
+      );
+      close();
+      return;
```

Comment:
See the PR description for this. No performance hit is expected as this boils down to basically 2 or 3 assembly instructions on almost every single architecture, ands the JIT will likely optimize it very well, as readyState is just an integer.